src/components/Header/HeaderButtons/index.tsx

```diff
@@ -116,82 +119,67 @@ const HeaderButtons = ({ setIsChooseNetwork }: IHeaderButtonsProps) => {
-      const timestampDiffInSeconds = Math.abs(
-        currentTimestamp.timestamp - subgraphTimestamp.timestamp
-      )
-      const diffInMinutes = Math.floor(timestampDiffInSeconds / 60)
-
-      const fiveMinutes = 5
-      const sixHoursInMinutes = 360
-      if (diffInMinutes <= fiveMinutes) {
-        return setCurrentSubgraphInfo({
-          network: chainInfo.network,
-          status: SubgraphStatus.Updated,
-          chainIcon: chainInfo.icon
-        })
-      }
-      if (diffInMinutes <= sixHoursInMinutes) {
-        return setCurrentSubgraphInfo({
-          network: chainInfo.network,
-          status: SubgraphStatus.PracticallyUpdated,
-          chainIcon: chainInfo.icon
-        })
-      }
+    const chaindId =
+      networks[Number(wallet?.chains[0].id ?? '')]?.chainId ?? 43114
 
-      setCurrentSubgraphInfo({
+    const provider = new JsonRpcProvider(networks[chaindId].rpc)
+    const subgraphTimestamp = await provider.getBlock(
+      BigInt(latestBlockData?.subgraphBlock)
+    )
+    const currentTimestamp = await provider.getBlock(
+      BigInt(latestBlockData?.currentBlock)
+    )
+
+    if (!(currentTimestamp?.timestamp && subgraphTimestamp?.timestamp)) {
+      return setCurrentSubgraphInfo({
         network: chainInfo.network,
-        status: SubgraphStatus.Outdated,
-        chainIcon: chainInfo.icon
+        status: SubgraphStatus.FetchingData,
+        chainIcon: chainInfo.icon,
+        diffTime: '0'
       })
     }
 
-    const blocksInFiveMinutes = 150
-    const blocksInSixHours = 10800
-    if (latestBlockData.diff <= blocksInFiveMinutes) {
+    const dataDiff = getDateDiff(
+      subgraphTimestamp.timestamp * 1000,
+      currentTimestamp.timestamp * 1000
+    )
+    const timestampDiffInSeconds = Math.abs(
+      currentTimestamp.timestamp - subgraphTimestamp.timestamp
+    )
+    const diffInMinutes = Math.floor(timestampDiffInSeconds / 60)
+
+    const fiveMinutes = 5
+    const sixHoursInMinutes = 360
+    if (diffInMinutes <= fiveMinutes) {
       return setCurrentSubgraphInfo({
         network: chainInfo.network,
         status: SubgraphStatus.Updated,
-        chainIcon: chainInfo.icon
+        chainIcon: chainInfo.icon,
+        diffTime: `${dataDiff?.value.toString().concat(' ', dataDiff?.string)}`
       })
     }
-    if (latestBlockData.diff <= blocksInSixHours) {
+
+    if (diffInMinutes <= sixHoursInMinutes) {
       return setCurrentSubgraphInfo({
         network: chainInfo.network,
         status: SubgraphStatus.PracticallyUpdated,
-        chainIcon: chainInfo.icon
+        chainIcon: chainInfo.icon,
+        diffTime: `${dataDiff?.value.toString().concat(' ', dataDiff?.string)}`
       })
     }
+
     setCurrentSubgraphInfo({
       network: chainInfo.network,
       status: SubgraphStatus.Outdated,
-      chainIcon: chainInfo.icon
+      chainIcon: chainInfo.icon,
+      diffTime: `${dataDiff?.value.toString().concat(' ', dataDiff?.string)}`
```

Comment:
Same thing here, you're already on a template literal, there's no need to use concat.